File: frontend/src/app/_models/auto-mod-plugin.ts
import {ModeratorRole} from './moderator-role';
import {Plugin} from './plugin';
import {BadWordsSettings} from './bad-words-settings';
import {RepeatedTextSettings} from './repeated-text-settings';
import {ServerInvitesSettings} from './server-invites-settings';
import {ExternalLinksSettings} from './external-links-settings';
import {ExcessiveCapsSettings} from './excessive-caps-settings';
import {ExcessiveEmojisSettings} from './excessive-emojis-settings';
import {ExcessiveSpoilersSettings} from './excessive-spoilers-settings';
import {ExcessiveMentionsSettings} from './excessive-mentions-settings';
import {ZalgoSettings} from './zalgo-settings';
import {AutomatedAction} from './automated-action';
import {ModeratorCommand} from './moderator-command';

export interface AutoModPlugin extends Plugin {
  badWordsSettings: BadWordsSettings;
  repeatedTextSettings: RepeatedTextSettings;
  serverInvitesSettings: ServerInvitesSettings;
  externalLinksSettings: ExternalLinksSettings;
  excessiveCapsSettings: ExcessiveCapsSettings;
  excessiveEmojisSettings: ExcessiveEmojisSettings;
  excessiveSpoilersSettings: ExcessiveSpoilersSettings;
  excessiveMentionsSettings: ExcessiveMentionsSettings;
  zalgoSettings: ZalgoSettings;
  automatedActions: AutomatedAction[];
  moderatorCommands: ModeratorCommand[];
  moderatorRoles: ModeratorRole[];
  ignoreBots: boolean;
}

File: backend/Models/AutoModPlugin.cs
using backend.Helpers;
using System.Collections.Generic;

namespace backend.Models
{
    public class AutoModPlugin : Plugin
    {
        public virtual BadWordsSettings BadWordsSettings { get; set; }
        public virtual RepeatedTextSettings RepeatedTextSettings { get; set; }
        public virtual ServerInvitesSettings ServerInvitesSettings { get; set; }
        public virtual ExternalLinksSettings ExternalLinksSettings { get; set; }
        public virtual ExcessiveCapsSettings ExcessiveCapsSettings { get; set; }
        public virtual ExcessiveEmojisSettings ExcessiveEmojisSettings { get; set; }
        public virtual ExcessiveSpoilersSettings ExcessiveSpoilersSettings { get; set; }
        public virtual ExcessiveMentionsSettings ExcessiveMentionsSettings { get; set; }
        public virtual ZalgoSettings ZalgoSettings { get; set; }
        public virtual IList<AutomatedAction> AutomatedActions { get; set; }
        public virtual IList<ModeratorCommand> ModeratorCommands { get; set; }
        public virtual List<ModeratorRole> ModeratorRoles { get; set; }
        public bool IgnoreBots { get; set; }
    }
}